### ISSUE TITLE
Support links without the # anonymiser

### ISSUE THREAD

**Issue by ara4n** [matrix-org/matrix.to#10]
The README says that https://matrix.to/@matthew:matrix.org and https://matrix.to/#matrix:matrix.org should work.  They don't.

**Comment by the-moog** [comment]
> The README says that https://matrix.to/@matthew:matrix.org and https://matrix.to/#matrix:matrix.org should work....

"They don't", rather they **_can't_** work. 👎  It may or may not be in the README, but that is at best an error if a conflicting specification already exists.

_However, what doing that does achieve is create a definitive answer on a Google search, e.g. Google "Canonically valid matrix id URL syntax" and you get this answer, which is incorrect and always has been.  If only Google had trained their AI/search results (same thing) on their own code!_


Client side entry validation on web forms often won't allow the `'@'` or `'#'` in the URL.
Even if it did, for an `http(s)` URL, anything after a` '#'` is locally interpreted, as specified in  [RFC3986](https://www.rfc-editor.org/rfc/rfc3986#section-3.5) & others. It defines the subsequent text as an `anchor` or `fragment`.

The `'#'` being interpreted locally turns the URL [https://matrix.to/#matrix:matrix.org](https://matrix.to/#matrix:matrix.org) into [https://matrix.to/](https://matrix.to/) when you click it, which is not going to work in any useful way.
and similarly (but more obscurely, coming down to implementation)
The `'@'`, if accepted at all, could then be perceived as an email separator, and hence get a `'mailto:'` prefixed by some web forms, creating an email address that then would be malformed, as there is an invalid domain separator character `':'`. 
i.e. [mailto:](mailto:) user="[https://matrix.to/](mailto:https://matrix.to/)" [\@](mailto:\@) domain="[matthew:matrix.org](mailto:matthew:matrix.org)" with an invalid DNS domain name.

Need proof?
Try entering the proposed URL, above, into **Contact Us** / **Links** section of a [YouTube](https://www.youtube.com/)  channel configuration.

Some webforms may allow you to use URL encoded  (aka escaped) forms for the non-premitted characters, some web forms block this as it is a way to break intended behaviour.
i.e.  This translation may or may not work.
`'@'` (User Sigil): %40
`'#'` (Room Sigil): %23
`':'` (Separator): %3A

Elsewhere a suggested workround (MSC2312) suggested many. But define none.

This method demands the validation accepts matrix: OR, if prefixed with https://matrix.to  that the input validation accepts a `':'` AND that it both accepts the `':'` OR, if not then accepts %3A
User Identifier: matrix:u/username:domain.tld
Room Alias: matrix:r/roomalias:domain.tld
Room ID: matrix:roomid/roomid:domain.tld
Event ID: matrix:roomid/roomid:domain.tld/e/eventid
NOTE: youtube.com does not.

This method demands the validation accept @ OR requires that the invalidation accepts escaped %NN inputs AND mandates that is does not further escape it, breaking the URL.
https://<homeserver>/_matrix/client/v3/user/@user:domain.tld
NOTE: youtube.com does not.

Maybe a better answer would be change the URL format accepted by the shortener at https://matrix.to ?
It could even accept multiple types, also make the matrix server itself accept shortened formats so that it does not become a collection point for private home server urls.
Then there is no need to mandate something that may or may not be accepted by input validation

Mandate that the input validation accepts any valid http complete URL, including the `'?'` argument specifier
https://matrix.to?user=matthew&server=matrix.org
or better assume the input validation ONLY accepts 'RESTFUL' type URLs
https://matrix.to/u/matthew/d/matrix.org
as equally and at the same time.
https://matrix.to/d/matrix.org/u/matthew

Ammusingly after spending half an hour trying to come up with a solution that does not requrie a change to matrix urls, Google's own AI said 

_I hear you. We've stripped away the "well-actually" spec talk and hit the cold, hard wall of reality.
To summarize the deadlock we've uncovered:
The Matrix Spec mandates characters (:, @, #) that are technically valid in a URI but "look like" garbage to 1990s-era security filters.
Web Validators (like YouTube) treat anything with a % or a : in the path as a potential exploit or a malformed string. The Result: 
There is no such thing as a canonically valid Matrix URL that is also "Strict-Validator Friendly."
By sticking to their specific sigil format, Matrix essentially opted out of being compatible with standard web form validation. 
You've correctly called out that the "solutions" usually offered are just different flavors of broken.
If you can't change the form and the protocol won't change its ID structure, the protocol is effectively invisible to that system._
